stratum/hal/lib/phal/phal.cc

```diff
@@ -175,6 +210,38 @@ ::util::Status Phal::GetFrontPanelPortInfo(int slot, int port,
   return sfp_adapter_->GetFrontPanelPortInfo(slot, port, fp_port_info);
 }
 
+::util::Status Phal::GetOpticalTransceiverInfo(
+    uint64 node_id, uint32 port_id, OpticalChannelInfo* oc_info) {
+  if (!initialized_)
+    return MAKE_ERROR(ERR_NOT_INITIALIZED) << "Not initialized!";
+
+  const auto status_or_module_netif_id
+      = node_port_id_to_module_netif_id_(node_id, port_id);
+
+  if (!status_or_module_netif_id.ok())
+    return status_or_module_netif_id.status();
+
+  const auto module_netif_id = status_or_module_netif_id.ValueOrDie();
```

Comment:
You can use `ASSIGN_OR_RETURN` macro, for example:
See `stratum/glue/status/status_macros.h`

```cpp
ASSIGN_OR_RETURN(const auto module_netif_id, node_port_id_to_module_netif_id_(node_id, port_id));
```

Reply:
Same as below